apps/dokuha-roadmap-frontend/src/app/details/[contentId]/page.tsx

```diff
@@ -0,0 +1,182 @@
+
+import Button from "@/components/ui/Button";
+
+export type postData = {
+  currentPage: number;
+  note: string;
+}
+
+export type getData = {
+  userId: string;
+  title: string;
+  totalPage: number;
+  currentPage: number;
+  note: string;
+}
+
+//TODO:alertの箇所はトースト通知ないしはonBlurでやりたい
+export default function DetailPage() {
+  const router = useRouter();
+
+  const { contentId } = useParams();
+  const [contentTitle, setContentTitle] = useState('');
+  const [currentPage, setCurrentPage] = useState(0);
+  const [totalPage, setTotalPage] = useState(0);
+  const [note, setNote] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  const progress = (currentPage / totalPage) * 100;
+  const clampedProgress = Math.max(0, Math.min(100, progress));
+
+  useEffect(() => {
+    const fetchContent = async () => {
+      setIsLoading(true);
+      try {
+        if (!contentId) {
+          return;
+        }
+
+        // TODO: URLにID直接指定することでほかのユーザのコンテンツが見れる＆編集できるので、clerkのトークン一緒に送ってサーバ側で検証する必要がある
+        const response = await fetch(`${process.env.NEXT_PUBLIC_BE_URL}learning-contents/${contentId}`);
+        if (!response.ok) {
+          throw new Error('コンテンツ情報の取得に失敗しました');
+        }
+
+        const data = await response.json();
+        const jsonResponse = data.data;
+
+        setContentTitle(jsonResponse.title);
+        setCurrentPage(jsonResponse.currentPage);
+        setTotalPage(jsonResponse.totalPage);
+        setNote(jsonResponse.note);
+      } catch (error) {
+        alert('エラーが発生しました。トップページに戻ります。');//TODO
+        router.push('/');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchContent();
+  }, [contentId, router]);
+
+  const onRecordSubmit = async (e: React.FormEvent) => {
+
+    // TODO:送信中はボタンを非活性にする。issubmittingみたいなステート
+    try {
+      e.preventDefault();
+      if (Number.isNaN(Number(currentPage)) || Number(currentPage) < 0) {
+        alert('現在のページ数は0以上の値を入力してください。');
+        return;
+      }
+
+      if (currentPage > totalPage) {
+        alert('現在のページ数は総ページ数よりも小さな値を入力してください。');
+        return;
+      }
+
+      const formatPOSTData: postData = {
+        "currentPage": Number(currentPage),
+        "note": note
+      }
+
+      const response = await fetch(`${process.env.NEXT_PUBLIC_BE_URL}learning-contents/${contentId}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(formatPOSTData),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      alert('記録が保存されました。');
+    } catch (error) {
+      // TODO: 適切なエラーハンドリングに置き換える
+      console.error('エラーが発生しました:', error);
+    }
+  };
+
+  const onBackPage = () => {
+    // TODO: 変更内容保存しないよ、みたいな警告だす。未編集の場合は出さないようにする？ 
+    router.push('/');
+  };
+
+  if (isLoading) {
+    return <div className="flex justify-center items-center h-screen">データ読み込み中...</div>;
+  }
+
```

Comment:
ここにこんな感じの関数を置くと少しだけスッキリしそう。切り出しておくと、単体テストを書いたりもしやすくなる
```suggestion
  }
  
  const getProgressMessage = (progress: number): string => {
    if (progress === 100) return 'D　O　K　U　H　A　！';
    if (progress >= 75) return 'もう少しで読了！がんばれ！';
    if (progress >= 25) return 'いい感じ！この調子で進めよう！';
    return '始まったばかり！コツコツ頑張ろう！';
  };

```